### DISCUSSION TITLE
Django-admin compatibility

### DISCUSSION THREAD

**Discussion by lc-thomas** [tortoise/tortoise-orm#1168] [Ideas]
TLDR; Why wouldn't you keep your models compatible with django-admin ?

First off, great project I've been looking around for a while for an acync django-like ORM for python, so I love this ♥

Then, one of my favorite thing with django's ORM, is the auto-made administration interface.
It's easily customizable with a few lines of code, and you get a great tool to manage permissions, check change log entries, and overall explore and manage your data. 
Django-admin has been an awesome time-saver in most of my projects.

Looking at your [Quick tutorial](https://github.com/tortoise/tortoise-orm#quick-tutorial), and the project description, you did a great work at keeping tortoise's models structure very similar to django's.
I mean except a few fields type's name changed (`IntegerField => IntField, ForeignKey => ForeignKeyField`), and the `on_delete` mandatory parameter missing in your `ForeignKeyField`, these models looks really like django's.

So what directly comes to my mind is, why wouldn't you have your models be compatible with django?
It would be amazing, you'd allow your users to be able to use django-admin with the same models.
And in the other direction, devs could easily switch from django's ORM to tortoise's in their projects.
Would just need to hack a little around to replace "tortoise" by "django" or vice-versa in those imports.
Also obviously django-admin queries wouldn't be async, but who cares really ?

Or is there anything else I'm missing?

**Comment by long2ice** [comment]
They two can't be compatibility, if you are finding something like django-admin, try https://github.com/fastapi-admin/fastapi-admin

**Comment by lc-thomas** [comment]
Of course they can be compatible, it's only a matter of using the same lexicon.
As I said earlier, there are only a few differences between a django model and tortoise's.
By the way I tried fastapi-admin, it's not as good as django-admin, the project is still quite young.

Then if tortoiseORM coders won't do compatibility with django's models, I'll probably end up coding a tortoise.models to django.models translator, because I'd really love to use tortoiseORM without letting go of django-admin.
